File: namastack-outbox-examples/namastack-outbox-example-stable-ids/src/main/java/io/namastack/demo/ExternalBroker.java
package io.namastack.demo;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class ExternalBroker {

    private static final Logger logger = LoggerFactory.getLogger(ExternalBroker.class);

    public static void publish(Object payload, String key) {
        logger.debug("[ExternalBroker] Publishing {} with key: {}", payload.getClass().getSimpleName(), key);
    }
}

File: namastack-outbox-examples/namastack-outbox-example-micrometer/src/main/kotlin/io/namastack/demo/ExternalBroker.kt
package io.namastack.demo

import io.namastack.demo.customer.CustomerRemovedEvent
import org.slf4j.LoggerFactory

object ExternalBroker {
    private val logger = LoggerFactory.getLogger(ExternalBroker::class.java)

    fun publish(event: CustomerRemovedEvent) {
        Thread.sleep((50..150).random().toLong())

        if (Math.random() < 0.4) { // 40% failure rate
            logger.warn("[External] Broker publish failed for key {}", event.id)
            throw RuntimeException("Simulated failure in ExternalBroker")
        }

        logger.info("[External] Event published with key {}", event.id)
    }
}
